### PR TITLE
vscode configuration (build task, code snippets)

### PR DESCRIPTION
**Pull request by EronWright**
## Description
This PR adds a vscode configuration consisting of:
- a content snippet for declaring a "chooser" block (use tabs to move from section to section).
- a default "build" task that runs "make build".

The `.gitignore` file is updated to allow for specific files within the `.vscode` folder, similar to https://github.com/pulumi/pulumi-kubernetes-operator/pull/491 and based on [github.com/github/gitignore/../VisualStudioCode.gitignore](https://github.com/github/gitignore/blob/main/Global/VisualStudioCode.gitignore).

https://github.com/pulumi/pulumi-hugo/assets/1775518/c55b3fba-d74c-47f9-aada-f9ac234187bb

https://github.com/pulumi/pulumi-hugo/assets/1775518/6ee7a848-4cbb-42e2-90c5-66f3d89e24e0

## Checklist:

- [ ] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [ ] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

File: .vscode/content.code-snippets [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+"{{% choosable language yaml %}}",
+"```yaml",
+"${6}",
+"```",
+"{{% /choosable %}}",
+"",
+"{{< /chooser >}}"
+        ]
+    }
+}
```
Comment (cnunciato) [inline .vscode/content.code-snippets:47]: We're trying to include JavaScript most everywhere now, so it'd be awesome if we could include that here as well as a nudge to do that. The trailing slash/self-closing mode is the preferred one for the chooser, as it causes it to act "globally" (i.e., to persist the selected language), and the empty lines around the code fences, while not always required, help keep the Markdown parser from going off the rails:

```suggestion
"{{< chooser language \"javascript,typescript,python,go,csharp,java,yaml\" />}}",
"",
"{{% choosable language javascript %}}",
"",
"```javascript",
"${1}",
"```",
"",
"{{% /choosable %}}",
"",
"{{% choosable language typescript %}}",
"",
"```typescript",
"${1}",
"```",
"",
"{{% /choosable %}}",
"",
"{{% choosable language python %}}",
"",
"```python",
"${2}",
"```",
"",
"{{% /choosable %}}",
"",
"{{% choosable language go %}}",
"",
"```go",
"${3}",
"```",
"",
"{{% /choosable %}}",
"",
"{{% choosable language csharp %}}",
"",
"```csharp",
"${4}",
"```",
"",
"{{% /choosable %}}",
"",
"{{% choosable language java %}}",
"",
"```java",
"${5}",
"```",
"",
"{{% /choosable %}}",
"",
"{{% choosable language yaml %}}",
"",
"```yaml",
"${6}",
"```",
"",
"{{% /choosable %}}",
"",
"{{< /chooser >}}"
        ]
    }
}
```

The `typescript` and `javascript` are just for consistency.

**Review by cnunciato — COMMENTED** [review]
Nice! Thank you for doing this. Just a couple of comments but otherwise looks great. ⭐

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+	// See https://go.microsoft.com/fwlink/?LinkId=733558
+	// for the documentation about the tasks.json format
+	"version": "2.0.0",
+	"tasks": [
+		{
+			"type": "shell",
+			"label": "make: build",
+			"command": "make build",
```
Comment (cnunciato) [inline .vscode/tasks.json:9]: FWIW, this is really only used in CI in practice; local development almost always uses `make serve`, and having a default task run `make build` somewhat suggests it's a thing that's commonly done. I don't use these VS Code tasks very often myself, though -- is there an analogous one that might go along with `make serve`? Maybe something debug-related? 🤔

Comment (EronWright) [reply .vscode/tasks.json:9]: Sorry for the delay, there's the concept of a default "build" task, and a default "test" task, and then one may have arbitrary tasks that you launch via the command bar. I don't think that "serve" would make most sense as a launch configuration of some sort.  This build task is harmless IMO but I'd be happy to remove it.

**Pull request closed without merging**